compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4436,7 +4460,7 @@ private boolean isAnydataOrJson(BType type) {
         return false;
     }
 
-    private BMapType getMapTypeForAnydataOrJson(BType type) {
+    private BMapType getMapTypeForAnydataOrJson(BType type, SymbolEnv env) {
```

Comment:
Why do we need this change?

Reply:
For certain cases, calling this method causes NPE, because `this.env` becomes null. I believe calling static functions leads to this NPE. Thats why i had to pass env as a parameter